### PR TITLE
added create item popup

### PR DESCRIPTION
**Pull request by aggarrohit**

### CODE REVIEW

File: src/components/CreateItem.jsx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            type="text"
+            onChange={(e) => setTitle(e.target.value)}
+          />
+
+          <label className="input-title">Description</label>
+          <textarea
+            className="input-area"
+            placeholder="product description.."
+            value={description}
+            type="text"
```
Comment (OlgaPinchuk) [inline src/components/CreateItem.jsx]: I think textarea does not have "type" attribute

File: src/components/CreateItem.jsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  };
+
+  return (
+    <div className="container">
+      <div className="create-item-container">
+        <div className="create-item-box">
+          <h2>Add Product</h2>
+
+          <label className="input-title">Title</label>
+          <input
```
Comment (OlgaPinchuk) [inline src/components/CreateItem.jsx:102]: I have added InputField component. Maybe it's better to refactor and use it instead? But it is up to you to decide

File: src/components/CreateItem.jsx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+      return;
+    }
+    const data = {
+      title: title,
+      description: description,
+      starting_price: startPrice,
+      end_time: selectedDate,
+      image: selectedImage,
+    };
+    console.log("item is : " + JSON.stringify(data));
```
Comment (OlgaPinchuk) [inline src/components/CreateItem.jsx]: If you do not need  console.log we can remove it

File: src/components/CreateItem.jsx [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    });
+
+    return result;
+  }
+
+  const handleAddButtonClick = () => {
+    if (!title) {
+      alert("Please enter title!");
+      return;
+    }
```
Comment (OlgaPinchuk) [inline src/components/CreateItem.jsx]: may be we can replace these if-checks with required attribute on inputs

File: src/components/CreateItem.jsx [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+              accept="image/png, image/jpeg"
+              onChange={(event) => onChange(event)}
+            />
+            <img
+              src={selectedImage || PlaceHolderImage}
+              onError={(event) => (event.currentTarget.src = PlaceHolderImage)}
+            />
+          </label>
+
+          <button className="bottom-button" onClick={handleAddButtonClick}>
```
Comment (OlgaPinchuk) [inline src/components/CreateItem.jsx]: It is better to handle onSubmit event on form but we need not to forget e.preventDefault() at the beginning of function body

File: src/styles/components/create-item-container.css [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: var(--space-small);
+}
+
+.select-time-button {
+  padding: var(--space-padding-small);
+}
+
+/* .input-image {
```
Comment (OlgaPinchuk) [inline src/styles/components/create-item-container.css]: Do you need this code in comments?

File: src/components/CreateItem.jsx [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    fetch(url, {
+      method: "POST",
+      body: JSON.stringify(data),
+      headers: {
+        Authorization: `Bearer ${user?.token}`,
+        "Content-Type": "application/json",
+      },
+    })
+      .then((response) => {
+        response.json().then((resp) => onSuccess(resp));
```
Comment (OlgaPinchuk) [inline src/components/CreateItem.jsx:82]: I can't find these functions (onSuccess, onFailure)

**Review by OlgaPinchuk — CHANGES_REQUESTED** [review]

**Review by rashmoni — APPROVED** [review]

File: src/components/CreateItem.jsx (hunk 1 above)
Comment (aggarrohit) [reply src/components/CreateItem.jsx]: Removed type from text-area

File: src/components/CreateItem.jsx (hunk 2 above)
Comment (aggarrohit) [reply src/components/CreateItem.jsx:102]: the create item form has 3 types of input fields so keeping as it is for now but will refactor in future

File: src/components/CreateItem.jsx (hunk 3 above)
Comment (aggarrohit) [reply src/components/CreateItem.jsx]: removed

File: src/components/CreateItem.jsx (hunk 4 above)
Comment (aggarrohit) [reply src/components/CreateItem.jsx]: done

File: src/components/CreateItem.jsx (hunk 5 above)
Comment (aggarrohit) [reply src/components/CreateItem.jsx]: done

File: src/styles/components/create-item-container.css (hunk 6 above)
Comment (aggarrohit) [reply src/styles/components/create-item-container.css]: removed

File: src/components/CreateItem.jsx (hunk 7 above)
Comment (aggarrohit) [reply src/components/CreateItem.jsx:82]: add these during API integration

**Review by OlgaPinchuk — APPROVED** [review]

**Review by AlirezaKToosi — APPROVED** [review]

**Pull request merged**